monkey/infection_monkey/exploit/zerologon.py

```diff
@@ -0,0 +1,377 @@
+"""
+Zerologon, CVE-2020-1472
+Implementation based on https://github.com/dirkjanm/CVE-2020-1472/ and https://github.com/risksense/zerologon/.
+"""
+
+import logging
+import os
+from binascii import unhexlify
+from typing import List, Optional
+
+import impacket
+from impacket.dcerpc.v5 import nrpc
+from impacket.dcerpc.v5.dtypes import NULL
+
+from common.utils.exploit_enum import ExploitType
+from infection_monkey.exploit.HostExploiter import HostExploiter
+from infection_monkey.exploit.zerologon_utils.dump_secrets import DumpSecrets
+from infection_monkey.exploit.zerologon_utils.options import \
+    OptionsForSecretsdump
+from infection_monkey.exploit.zerologon_utils.wmiexec import Wmiexec
+from infection_monkey.network.zerologon_fingerprint import ZerologonFinger
+from infection_monkey.utils.capture_output import StdoutOutputCaptor
+
+LOG = logging.getLogger(__name__)
+
+
+class ZerologonExploiter(HostExploiter):
+    _TARGET_OS_TYPE = ['windows']
+    _EXPLOITED_SERVICE = 'Netlogon'
+    EXPLOIT_TYPE = ExploitType.VULNERABILITY
+    RUNS_AGENT_ON_SUCCESS = False
+    MAX_ATTEMPTS = 2000
+    ERROR_CODE_ACCESS_DENIED = 0xc0000022
+
+    def __init__(self, host: object):
+        super().__init__(host)
+        self.vulnerable_port = None
+        self.zerologon_finger = ZerologonFinger()
+        self.exploit_info['credentials'] = {}
+
+    def _exploit_host(self) -> Optional[bool]:
```

Comment:
Does it require `Optional`? I think it would be better to always return either True or False